actors/builtin/paych/paych_actor.go

```diff
@@ -57,23 +58,21 @@ func (pca *Actor) Constructor(rt vmr.Runtime, params *ConstructorParams) *adt.Em
 }
 
 // Resolves an address to a canonical ID address and requires it to address an account actor.
-// The account actor constructor checks that the embedded address is associated with an appropriate key.
-// An alternative (more expensive) would be to send a message to the actor to fetch its key.
-func (pca *Actor) resolveAccount(rt vmr.Runtime, raw addr.Address) (addr.Address, error) {
-	resolved, ok := rt.ResolveAddress(raw)
-	if !ok {
-		return addr.Undef, exitcode.ErrNotFound.Wrapf("failed to resolve address %v", raw)
+func (pca *Actor) resolveAccount(rt vmr.Runtime, raw addr.Address) (addr.Address, error, exitcode.ExitCode) {
+	resolved, err := builtin.ResolveToIDAddr(rt, raw)
+	if err != nil {
+		return addr.Undef, xerrors.Errorf("failed to resolve address %v: %w", raw, err), exitcode.ErrIllegalState
 	}
 
 	codeCID, ok := rt.GetActorCodeCID(resolved)
 	if !ok {
-		return addr.Undef, exitcode.ErrForbidden.Wrapf("no code for address %v", resolved)
+		return addr.Undef, xerrors.Errorf("no code for address %v", resolved), exitcode.ErrIllegalArgument
```

Comment:
`exitcode.ErrIllegalArgument.Wrapf(...)`

Reply:
This is done.